src/lib.rs

```diff
@@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     marker: marker::PhantomData<(&'a K, &'a mut V)>,
 }
 
+/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
+pub struct IntoIter<K, V, S> {
```

Comment:
You can mem::forget the map.

On Friday, April 22, 2016, David Tolnay <notifications@github.com> wrote:

> In src/lib.rs
> <https://github.com/contain-rs/linked-hash-map/pull/50#discussion_r60803631>
> :
>
> > @@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
> >      marker: marker::PhantomData<(&'a K, &'a mut V)>,
> >  }
> >
> > +/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
> > +pub struct IntoIter<K, V, S> {
>
> No because we rely on the map to own every Box<LinkedHashMapEntry<K, V>>.
> If we don't keep the map, we need to move the data out of it first, or
> rework how the whole thing is represented so that the map no longer owns
> the data.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/contain-rs/linked-hash-map/pull/50/files/6bf364b7bc09322a7f900536f1e78bd8d03904be#r60803631>
>